src/python/hooks/README.md

```diff
@@ -0,0 +1,96 @@
+# Overview
+
+The python scripts in this folder are intended to be run in a python-based
+`pre-commit` harness that facilitates running custom python scripts in
+`pre-commit`, both for developers locally and in CI with github actions.
+
+If you are contributing to the python files in this repository for the first
+time, make sure to [install the required tools first].
+
+If you want to quickly format your python files so that they pass the
+`pre-commit` checks, you can [run our formatting script].
+
+## Running `pre-commit` hooks locally
+
+To avoid your PR contributions failing the CI pipeline checks, you can install
+the `pre-commit` hooks locally so that they automatically run right before you
+add a new commit.
+
+### Installing the required tools
+
+To properly install the full `pre-commit` environment for this repository,
+you'll need the following command-line tools:
+
+- `git`, `python`, and `pip` (if you're not using `brew`)
+- `poetry` and `pre-commit`
+
+You can install all of these tools with your preferred package manager.
+If you have `python` and `pip` installed already, you might use:
+
+- `pip install poetry`
+- `pip install pre-commit`
+
+If you use `brew` on MacOS:
+
+- `brew install poetry`
+- `brew install pre-commit`
+
+### Installing the `poetry` and `pre-commit` environments
+
```

Comment:
Generally I think this content should go in `cfg/README.md`

I get that since a lot of it is pertinent to the Python hooks you've chosen to add docs in the `src/python/hooks` dir, but I think it is a bit confusing that the general `pre-commit` content is nested down here. It might be worth making a separate README for the Python hooks and one for `pre-commit` generally, but at the very least probably something in `cfg/README.md` would be useful

Reply:
ah yes I see what you're saying, a lot of this is more about using `pre-commit` than the actual scripts here. I'll separate the relevant content into two `README`s and point them to each other in subsections